lib/compat/wordpress-5.9/theme-templates.php

```diff
@@ -72,3 +72,16 @@ function gutenberg_filter_wp_template_unique_post_slug( $override_slug, $slug, $
 // Remove 5.8 filter if existant.
 remove_filter( 'pre_wp_unique_post_slug', 'wp_filter_wp_template_unique_post_slug' );
 add_filter( 'pre_wp_unique_post_slug', 'gutenberg_filter_wp_template_unique_post_slug', 10, 5 );
+
+/**
+ * Enable block templates (editor mode) for themes with theme.json.
+ */
+function gutenberg_enable_block_templates() {
+	if ( wp_is_block_theme() || WP_Theme_JSON_Resolver_Gutenberg::theme_has_support() ) {
```

Comment:
In this function I added `wp_is_block_theme` to auto-enable block-templates for block themes. It was already the case in the Gutenberg plugin because before this PR, the plugin used `gutenberg_supports_block_templates` to check support but for core, if a block theme didn't have a "theme.json" file, block templates support would have been disabled. So this change here need to be backported into Core.

Reply:
Looks like you already made this change in Core so I'll do nothing 😀

https://github.com/WordPress/wordpress-develop/commit/f7d2a2ee9d003633e6c729c0835cd4addd23f9b3